rnd/autogpt_libs/autogpt_libs/logging/config.py

```diff
@@ -0,0 +1,199 @@
+            return v
+        return v
+
+
+def configure_logging(
+        force_cloud_logging: bool = False
+) -> None:
+    """Configure the native logging module based on the LoggingConfig settings.
+
+    This function sets up logging handlers and formatters according to the
+    configuration specified in the LoggingConfig object. It supports various
+    logging outputs including console, file, cloud, and JSON logging.
+
+    The function uses the LoggingConfig object to determine which logging
+    features to enable and how to configure them. This includes setting
+    log levels, log formats, and output destinations.
+
+    No arguments are required as the function creates its own LoggingConfig
+    instance internally.
+
+    Note: This function is typically called at the start of the application
+    to set up the logging infrastructure.
+    """
+
+    config = LoggingConfig()
+
+    log_handlers: list[logging.Handler] = []
+
+    # Cloud logging setup
+    if config.enable_cloud_logging or force_cloud_logging:
+        try:
+            import google.cloud.logging
+            from google.cloud.logging.handlers import CloudLoggingHandler
+            from google.cloud.logging_v2.handlers.transports.sync import SyncTransport
+
+            client = google.cloud.logging.Client()
+            cloud_handler = CloudLoggingHandler(
+                client,
+                name="autogpt_logs",
+                transport=SyncTransport,
+            )
+            cloud_handler.setLevel(config.level)
+            cloud_handler.setFormatter(StructuredLoggingFormatter())
+            log_handlers.append(cloud_handler)
+            print("Cloud logging enabled")
+        except ImportError:
+            print(
+                "Cloud logging requested but google-cloud-logging is not installed. "
+                "Please install it with 'pip install google-cloud-logging'."
+            )
+        except Exception as e:
+            print(f"Failed to set up cloud logging: {str(e)}")
+
+    # File logging setup
+    if config.enable_file_logging:
+        # create log directory if it doesn't exist
+        if not config.log_dir.exists():
+            config.log_dir.mkdir(parents=True, exist_ok=True)
+
+        # Activity log handler (INFO and above)
+        activity_log_handler = logging.FileHandler(
+            config.log_dir / LOG_FILE, "a", "utf-8"
+        )
+        activity_log_handler.setLevel(config.level)
+        activity_log_handler.setFormatter(
+            AGPTFormatter(SIMPLE_LOG_FORMAT, no_color=True)
+        )
+        log_handlers.append(activity_log_handler)
+
+        # Debug log handler (all levels)
+        debug_log_handler = logging.FileHandler(
+            config.log_dir / DEBUG_LOG_FILE, "a", "utf-8"
+        )
+        debug_log_handler.setLevel(logging.DEBUG)
+        debug_log_handler.setFormatter(AGPTFormatter(DEBUG_LOG_FORMAT, no_color=True))
+        log_handlers.append(debug_log_handler)
```

Comment:
I'm not sure this should be enabled by default if file logging is enabled; debug logging in our current system generates a LOT of output.